includes/class-slatwall-integration.php

```diff
@@ -27,359 +27,350 @@ private function getKeyValue(){
-                curl_close($ch);
-            } catch(Exception $e) {
-
-            if($e->getCode() !== 6){
-              $content = trigger_error(sprintf(
-                    'Curl failed with error #%d: %s',
-                    $e->getCode(), $e->getMessage()),
-                    E_USER_ERROR);
-                    } else {
-                      $error_msg = $e->getMessage();
-                            $content = '{"errors" : "'.$error_msg.'"}';
-                    }
-
             }
             $content_obj = json_decode($content);
+            $cookies = (object) $cookies1;
             $content_obj->cookies = $cookies;
+           // d($content_obj);
              return $content_obj;
+                } else {
+                    return false;
+                }
            } else {
                return false;
            }
 
         }
 
         protected function post_API_integration(array $request,string $API_URL,string $method = 'POST'){
-            $auth = AUTHORIZATION;
+            $auth = SLATWALL_AUTHORIZATION;
             $key_data = $this->getKeyValue();
             if($key_data){
             $token = $key_data->token;
             unset($request['registration']);
             $post_field_data = $request;
             $domain = $key_data->domain;
             $full_api_url = $domain.$API_URL;
-            //print_r($post_field_data);
-            $curl = curl_init();
-
-            curl_setopt_array($curl, array(
-              CURLOPT_URL => $full_api_url,
-              CURLOPT_RETURNTRANSFER => true,
-              CURLOPT_ENCODING => "",
-              CURLOPT_MAXREDIRS => 10,
-              CURLOPT_TIMEOUT => 0,
-              CURLOPT_FOLLOWLOCATION => true,
-              CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
-              CURLOPT_CUSTOMREQUEST => $method,
-              CURLOPT_SSL_VERIFYHOST => FALSE,
-              CURLOPT_SSL_VERIFYPEER => FALSE,
-              CURLOPT_POSTFIELDS => $post_field_data,
-              CURLOPT_HTTPHEADER => array(
-                "Auth-Token: Bearer $token",
-                "Authorization: Basic $auth"
-              ),
-            ));
-
-            $response = curl_exec($curl);
-
-            curl_close($curl);
-           // print_R($response); die("registration");
+            $http_header = array();
+            if($token != ""){
+              $http_header["Auth-Token"] = "Bearer ".$token;
+            } 
+            if($auth){
+                $http_header["Authorization"] = "Basic ".$auth;
+            }
+            $post_field_arg = array('returntransfer'=>true,
+                    'encoding'=>'',
+                    'maxredirs'=>10,
+                    'verbose'=>1,
+                   'body'=>$post_field_data,
+                    'followlocation'=>true,
+                 'timeout' => 10,
+                'header'=>1,
+                    'headers' => $http_header
+                    
+                    );
```

Comment:
@raj-yash let's move this to a separate function. that way we can generate payload for both GET and POST methods from a single method.